### ISSUE TITLE
Requirements to build Pytorch on Windows

### ISSUE THREAD

**Issue by ZmeiGorynych** [pytorch/pytorch#6109]
Hi,

I'm trying to follow the [instructions](https://github.com/pytorch/pytorch#from-source) to build pytorch on Windows (10 Pro), and am encountering the following issues:
* `conda install` can't find `mkl-include`
* Is Visual Studio 17 Enterprise required? I dowloaded the Community Edition, but then `call "%VS150COMNTOOLS%\vcvarsall.bat" x64 -vcvars_ver=14.11` fails.
* Where can I download/install NVTX? Googling and searching on developer.nvidia.com didn't bring any results.

Thanks a lot,
E.

**Comment by peterjc123** [comment]
1. Would you please update your conda using `conda update conda`?
2. No, Community version is enough. I guess your version of MSVC BuildTools is too high or the PATH of VS150COMNTOOLS may be incorrect.
3. Make sure you install CUDA after installing the corresponding version of MSVC. And it will be installed. Actually it is located in the setup executable. You can extract it and find that.

**Comment by ZmeiGorynych** [comment]
Thanks a lot for replying!
1. Was resolved after following your instructions.
2.  Turns out the VS installer with default settings doesn't install the compiler; after installing everything that looked likely from the VS IDE, and changing 'Enterprise' for 'Community' in the instructions, and changing `vcvars_ver` to 14.13 in the command below ( I did a search for '14' in the VS install directory, and there were a couple of directories called 14.13.something, so I tried that and it worked), the script completed without issues.
I installed CUDA 9.0 and cuDNN 7 (is CUDA 9.1 or 9.2 supported btw?), and tried running 
`python setup.py install`

That got off to a good start (full output below) but then got stuck at 
```
c:\program files\nvidia gpu computing toolkit\cuda\v9.0\include\crt/host_config.h(133): fatal error C1189: #error:  --
unsupported Microsoft Visual Studio version! Only the versions 2012, 2013, 2015 and 2017 are supported! [C:\Users\egork
\Dropbox\GitHub\pytorch\aten\build\src\ATen\ATen.vcxproj]
```
I then looked at host_config.h from the message above, and edited line 131 to read
`#if _MSC_VER < 1600 || _MSC_VER > 1913` instead of 1911

That fixed the above error, but then it started raining exceptions like this:
```
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.13.26128\include\type_traits(506): error : expression must have a constant value [C:\Users\egork\Dropbox\GitHub\pytorch\aten\build\src\ATen\ATen.vcxproj]
```
Any idea how to fix that? I do have VC17, see the About box below
![image](https://user-images.githubusercontent.com/22304254/38208668-c074e50c-36a9-11e8-8bcb-6e7e56379b4d.png)
